File: src/actions/plan.go
package actions

import "enva/libs"

// DeployPlan holds deployment sequencing information
type DeployPlan struct {
	AptCacheContainer *libs.ContainerConfig
	Templates         []*libs.TemplateConfig
	ContainersList    []*libs.ContainerConfig
	TotalSteps        int
	Step              int
	StartStep         int
	EndStep           *int
	CurrentActionStep int
	PlanOnly          bool
}

File: src/Actions/Plan.cs
using Enva.Libs;

namespace Enva.Actions;

public class DeployPlan
{
    public ContainerConfig? AptCacheContainer { get; set; }
    public List<TemplateConfig> Templates { get; set; } = new();
    public List<ContainerConfig> ContainersList { get; set; } = new();
    public int TotalSteps { get; set; }
    public int Step { get; set; }
    public int StartStep { get; set; }
    public int? EndStep { get; set; }
    public int CurrentActionStep { get; set; }
    public bool PlanOnly { get; set; }
}
